src/components/Api.js

```diff
@@ -0,0 +1,90 @@
+export default class Api {
+  constructor({ baseUrl, authorization }) {
+    this._baseUrl = baseUrl;
+    this._authorization = authorization;
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: {
+        authorization: this._authorization,
+      },
+    });
```

Comment:
According to the task you need to check the response if it's ok and **throw** an error if it's not ok. And this `ok` checking should be only in `Api` to clean up the code of other files. It's the responsibility of `Api`


```js
.then(res => {
                if (res.ok) {
                    return res.json();
                }
                return Promise.reject(`Error ${res.status}`);
            })

```

The code of `ok` checking is repeated in **every** request of the project. To avoid it you need to make a special method `_checkResponse` that will check any response from the server if it’s ok or not.


```js
_checkResponse(res) {
        // here is the code of the checking
    }

```

And now you can replace redundant repeating with the only line:


```js
 .then(this._checkResponse)

```

Please, pay attention that you need to pass only the `reference` for the method rather than the call. 

**Could be improved**

You can make a special method for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.

Reply:
Does Promise.reject() here satisfy your comment for catching errors at the end of every request?

Reply:
> Does Promise.reject() here satisfy your comment for catching errors at the end of every request?

`Promise.reject()` throws an error (it doesn't catch anything)